scipy/optimize/_trustregion_iterative.py

```diff
@@ -0,0 +1,470 @@
+"""Iterative trust-region optimization subproblem."""
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from scipy.linalg import norm, get_lapack_funcs, solve_triangular, norm
+from ._trustregion import (_minimize_trust_region, BaseQuadraticSubproblem)
+
+__all__ = ['estimate_smallest_singular_value',
+           'singular_leading_submatrix',
+           'IterativeSubproblem']
+
+
+def _minimize_trust_iterative(fun, x0, args=(), jac=None, hess=None,
+                              **trust_region_options):
+    """
+    Minimization of scalar function of one or more variables using
+    a iterative trust-region algorithm.
+
+    Options
+    -------
+    initial_trust_radius : float
+        Initial trust-region radius.
+    max_trust_radius : float
+        Maximum value of the trust-region radius. No steps that are longer
+        than this value will be proposed.
+    eta : float
+        Trust region related acceptance stringency for proposed steps.
+    gtol : float
+        Gradient norm must be less than ``gtol` before successful
+        termination.
+    trust_region_factor, optimization_factor : float
+        Parameters that determines the stop criteria for the subproblem solver.
+        The iterative algoritm for the trust-region subproblem will stop when
+        the computed step ``p`` fullfil
+        ``norm(p)<=trust_region_factor*trust_radius``
+        or when the solution ``m`` for the subproblem fullfil
+        ``m < optimization_factor*m_optimal``, with ``m_optimal`` being the
```

Comment:
> I am thinking in letting k_easyand k_hard as constructor parameters of the class IterativeSubproblem. It would make this class much easier to test and I think would not affect the code legibility.

Looks reasonable to me as well. Change it and we can move on.